plugins/base/src/main/kotlin/renderers/html/HtmlRenderer.kt

```diff
@@ -49,16 +54,26 @@ open class HtmlRenderer(
     ) {
         div("platform-hinted") {
             attributes["data-platform-hinted"] = "data-platform-hinted"
-            val contents = nodes.toList().mapIndexed { index, (sourceSet, elements) ->
-                sourceSet to createHTML(prettyPrint = false).div(classes = "content") {
-                    if (index == 0) attributes["data-active"] = ""
-                    attributes["data-togglable"] = sourceSet.sourceSetName
+            var counter = 0
+            val contents = nodes.toList().map { (sourceSet, elements) ->
+                sourceSet to createHTML(prettyPrint = false).div {
                     elements.forEach {
                         buildContentNode(it, pageContext, setOf(sourceSet))
                     }
+                }.stripDiv()
+            }.groupBy(Pair<SourceSetData, String>::second, Pair<SourceSetData, String>::first).entries.flatMap { (html, sourceSets) ->
+                sourceSets.filterNot {
+                    sourceSetDependencyMap[it].orEmpty().any { dependency -> sourceSets.contains(dependency) }
+                }.map {
+                    it to createHTML(prettyPrint = false).div(classes = "content") {
+                        if (counter++ == 0) attributes["data-active"] = ""
```

Comment:
```suggestion
                        if (index == 0) attributes["data-active"] = ""
```

Reply:
Nope, this is a counter to flatMap, not this map. Your solution will mark as "data-active" every unique string in tabs